### PR TITLE
Invitation dialog 18b

### PR DESCRIPTION
**Pull request by cch5ng**
majority of styles were fixed. I'm still having challenges with the width sizing of the text inputs. they are a little too wide. I tried to restyle DialogContent but was having alignment issues.

I did not install the cors package but can do that part in an integration ticket.

------

to test:

* the application needs to be in authenticated state (currently default but may change after auth changes are merged)

* from the left pane, click the Invite Friends button (under Friends tab)

* email input should handle either single email or list of emails (comma separated). currently there is validation for empty email or invalid email values but the error is not displayed. the error validation is possibly too broad (failing on any invalid email in the list).

* the copy button should work (currently the url is hard coded)

* the submit button should not work (cors was added in a different ticket and also the BE requires refactoring) but an http request should be made (should expect to fail)

* the close button (upper right) should also work

### CODE REVIEW

**Comment by sina-jamshidi** [comment]
Looks great Carol! You can keep the styling as is, but if we find some time later and you want to play around with it, you can try my suggestion below.

> majority of styles were fixed. I'm still having challenges with the width sizing of the text inputs. they are a little too wide. I tried to restyle DialogContent but was having alignment issues.

I assume you already tried putting your DialogContent in a <Container> to make them more centred and less wide. That seems to mess with your copy button though. One way around this is to use an <InputAdornment> instead. You can change your InputProps on the <TextField> to be something like:

`              InputProps={{
                readOnly: true,
                endAdornment: (
                  <InputAdornment position="end">
                    <IconButton
                      size="small"
                      color="primary" 
                      edge="start"
                      disableRipple={true}
                      value={referralUrl}
                      onClick={handleClickCopyBtn}
                    >
                  Copy Link
                    </IconButton>
                  </InputAdornment>),
              }}`

Some demos of adornments and icons [here](https://material-ui.com/components/text-fields/#icons). Feel free to take some liberties with the mockups (for example maybe if you can't figure out the best styling, you can do something else like use FileCopyIcon, or put the button directly beside the textfield).

> * email input should handle either single email or list of emails (comma separated). currently there is validation for empty email or invalid email values but the error is not displayed. the error validation is possibly too broad (failing on any invalid email in the list).

That validation scope is fine, you can fail on any invalid email.

File: client/src/components/Invitations/InvitationDialog.js [hunk 1]
```diff
@@ -0,0 +1,281 @@
...
+      } else {
+        for (let i = 0; i < emailAr.length; i++) {
+          if (!isEmailValid(emailAr[i])) {
+            emailsAreValid = false;
+            break;
+          }
+        }  
+      }
+
+      if (!emailsAreValid) {
```
Comment (bonnieli) [inline client/src/components/Invitations/InvitationDialog.js]: Suggestion to rewrite the logic to reduce repeated code:

```

const emailsAreValid = (emailArrays) => {
  emailArrays.forEach( (email) => { 
    if (!isEmailValid(email) return false;
  })

  return true;
}

if (!email.length) {
} else if (!emailsAreValid(emailArray) {
} else {
  // send to backend, by this point it should have passed all validation test
}
```

Comment (bonnieli) [reply client/src/components/Invitations/InvitationDialog.js]: so even if email Array is length 1, going through that helper function will work

Comment (cch5ng) [reply client/src/components/Invitations/InvitationDialog.js]: done. that makes sense.

**Comment by cch5ng** [comment]
> Looks great Carol! You can keep the styling as is, but if we find some time later and you want to play around with it, you can try my suggestion below.

thanks @sina-jamshidi , I tried your container suggestion and the result did not look bad (even with the copy button outside of the text input) but I felt like the text input became a little too narrow for a list of emails. so I am leaving this as is. the icon method sounded like it could be more confusing for the user or take a bit more time. but I will keep it mind for the future.

File: client/src/components/Invitations/InvitationDialog.js (hunk 1 above)
Comment (bonnieli) [reply client/src/components/Invitations/InvitationDialog.js]: revised code looks good!

**Pull request merged**